app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -54,6 +54,10 @@ def uuid_from_sp_identity(identity)
   end
 
   def email_from_sp_identity
+    if identity.email_address_id &&
+       (email_address = EmailAddress.find_by(id: identity.email_address_id))
+      return email_address.email
+    end
     email_context.last_sign_in_email_address.email
```

Comment:
I had a [better suggestion in my earlier comment](https://github.com/18F/identity-idp/pull/10951#discussion_r1725070439), would it work?

```suggestion
    identity.email_address&.email || email_context.last_sign_in_email_address.email
```

Reply:
I get a JSON::ParserError with that. Maybe I'm missing an obvious tweak that would make it work.
![Screenshot 2024-08-21 at 12 11 09 PM (2)](https://github.com/user-attachments/assets/5423e339-c889-4f3b-92a9-c6517a245886)

Reply:
I think the issue is that we've setup the relationship incorrectly between Identity and EmailAddress, the difference between [`belongs_to`](https://guides.rubyonrails.org/association_basics.html#the-belongs-to-association) and [`has_one`](https://guides.rubyonrails.org/association_basics.html#the-has-one-association) being that assigning `has_one` on Identity to EmailAddress would assume that EmailAddress has a one-to-one relationship to Identity through a property on the `email_addresses` table (the `service_provider_identity_id` in the error you showed.

Instead, we want a one-to-many relationship between Identity and EmailAddress.

I think this should work, which I tested locally as well to verify fix to both the error you shared, as well as still working as expected when destroying the linked email address:

```diff
diff --git a/app/models/email_address.rb b/app/models/email_address.rb
index 0f4ca204a6..4ee1814e4e 100644
--- a/app/models/email_address.rb
+++ b/app/models/email_address.rb
@@ -13,2 +13,3 @@ class EmailAddress < ApplicationRecord
   # rubocop:enable Rails/HasManyOrHasOneDependent
+  has_many :identities, class_name: 'ServiceProviderIdentity', dependent: :nullify
 
diff --git a/app/models/service_provider_identity.rb b/app/models/service_provider_identity.rb
index c7d49ec9d1..5030aea024 100644
--- a/app/models/service_provider_identity.rb
+++ b/app/models/service_provider_identity.rb
@@ -21,4 +21,3 @@ class ServiceProviderIdentity < ApplicationRecord
 
-  has_one :email_address,
-          dependent: :nullify
+  belongs_to :email_address
```